### PR TITLE
Update NGP related models with nerfacc 0.5.2

### PR DESCRIPTION
**Pull request by liruilong940607**
This PR separates nerfacc update from the PR #1705. It bumps the `nerfacc` dependence to its latest version (0.5.2), and updates NGP related models (`instant-ngp`, `instant-ngp-bounded`, `nerfplayer-ngp`) with the latest nerfacc. 

Note on the master branch, for some reason the [non-gradient skipping](https://github.com/nerfstudio-project/nerfstudio/blob/9b99802042386b5ae2844b1ddccb366ce2a2f24d/nerfstudio/model_components/ray_samplers.py#L472-L473) is disabled with nerfacc. This PR re-enables that with the latest nerfacc.

After this PR, we can update the docs for the ngp-based methods, such as #1774.

### CODE REVIEW

**Comment by liruilong940607** [comment]
`instant-ngp-bounded` model is tested on nerf-synthetic lego:

```bash
CUDA_VISIBLE_DEVICES=3 ns-train instant-ngp-bounded --data ./data/nerf_synthetic/lego/transforms_train.json --vis wandb --max-num-iterations 10000
```

Screen Shot 2023-04-24 at 4 18 59 PM

Screen Shot 2023-04-24 at 4 18 31 PM

Screen Shot 2023-04-24 at 4 18 03 PM

Screen Shot 2023-04-24 at 4 17 43 PM

**Comment by liruilong940607** [comment]
`instant-ngp` is tested with the `poster` scene:

```bash
CUDA_VISIBLE_DEVICES=3 ns-train instant-ngp --data data/nerfstudio/poster --vis wandb --max-num-iterations 10000
```
Screen Shot 2023-04-24 at 4 35 20 PM
Screen Shot 2023-04-24 at 4 35 41 PM
Screen Shot 2023-04-24 at 4 36 05 PM
Screen Shot 2023-04-24 at 4 35 54 PM

**Comment by liruilong940607** [comment]
`nerfplayer-ngp` is tested with `dycheck`'s sequence `mochi-high-five`:

```bash
CUDA_VISIBLE_DEVICES=3 ns-train nerfplayer-ngp --vis wandb --max-num-iterations 10000 --data ./data/mochi-high-five/
```

Screen Shot 2023-04-24 at 4 38 03 PM
Screen Shot 2023-04-24 at 4 38 15 PM
Screen Shot 2023-04-24 at 4 38 43 PM
Screen Shot 2023-04-24 at 4 38 33 PM
Screen Shot 2023-04-24 at 4 38 26 PM

**Review by tancik — APPROVED** [review]
LGTM, thanks for updating. The results look promising!

**Comment by liruilong940607** [comment]
In summary:
- for `instant-ngp-bounded` the metrics match with the original one with ~3x faster training.
- for `instant-ngp` the master version somehow fails to converge. I'm not sure why but now it's good.
- for `nerfplayer-ngp` the metrics are consistently better, with 1.5x training time. But considering `num_rays_per_batch` is 3x than before, the 1.5x training time is reasonable.

The major difference is that now we re-enable the [non-gradient skipping](https://github.com/nerfstudio-project/nerfstudio/blob/9b99802042386b5ae2844b1ddccb366ce2a2f24d/nerfstudio/model_components/ray_samplers.py#L472-L473) for the occluded regions in the sampling procedure.

File: nerfstudio/models/instant_ngp.py [hunk 1]
```diff
@@ -102,28 +104,33 @@ def populate_modules(self):
         """Set the fields and modules."""
         super().populate_modules()
 
-        self.field = TCNNInstantNGPField(
+        if self.config.disable_scene_contraction:
+            scene_contraction = None
+        else:
+            scene_contraction = SceneContraction(order=float("inf"))
```
Comment (hturki) [inline nerfstudio/models/instant_ngp.py:110]: @liruilong940607 is scene contraction meant to be used with the new multi-level grid? Should TCNNNerfactoField just not take the expanded aabb as in https://github.com/KAIR-BAIR/nerfacc/blob/master/examples/train_ngp_nerf_occ.py#L130

Comment (liruilong940607) [reply nerfstudio/models/instant_ngp.py:110]: The field and multi-level occ grid are completely separated with each other. The new occ grid does not support contraction anymore but the field can still do contraction. I didn't play around with the field to compare it with/without contraction so I just kept it as is. 

Plz share if you find anything!

**Pull request merged**